spec/rails_spec_helper.rb

```diff
@@ -28,6 +28,10 @@ def generate
       config.logger = fake_logger
       Rails.application.routes.default_url_options = {host: "example.com"}
 
+      if Rails::VERSION::MAJOR >= 6
+        Rails.application.config.hosts << "www.example.com"
```

Comment:
New security measure in rails 6. From the [changelog](https://github.com/rails/rails/blob/5e6e505083c2b0caf85b2f86c6be3ff3d8750857/railties/CHANGELOG.md):

> The ActionDispatch::HostAuthorization is a new middleware that prevent against DNS rebinding and other Host header attacks. It is included in the development environment by default with the following configuration:

```ruby
Rails.application.config.hosts = [
  IPAddr.new("0.0.0.0/0"), # All IPv4 addresses.
  IPAddr.new("::/0"),      # All IPv6 addresses.
  "localhost"              # The localhost reserved domain.
]
```

Without this setting, request specs fail in rails 6 due to our setting example.com as the default URL host.